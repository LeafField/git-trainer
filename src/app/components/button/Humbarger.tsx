import React, { FC } from "react";
import Question from "../icons/Question";
import Cross from "../icons/Cross";

type Props = {
  isHidden: boolean;
  HumbargerOpen: () => void;
};

const Humbarger: FC<Props> = ({ HumbargerOpen, isHidden }) => {
  return (
    <button
      className="absolute right-1 top-1 z-50 block w-fit lg:hidden"
      aria-controls="aside"
      aria-expanded={isHidden ? "false" : "true"}
      onClick={HumbargerOpen}
      data-testid="humbarger"
    >
      {isHidden ? <Question /> : <Cross />}
    </button>
  );
};

export default Humbarger;
