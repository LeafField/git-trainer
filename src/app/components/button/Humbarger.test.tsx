import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Humbarger from "./Humbarger";

describe("Humbargerの単体テスト", () => {
  it("ドロワーメニューが閉じている時に正しくレンダリングされているか", () => {
    const { container } = render(
      <Humbarger HumbargerOpen={() => {}} isHidden />,
    );
    expect(screen.getByTestId("humbarger")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("ドロワーメニューが開いている時に正しくレンダリングされているか", () => {
    const { container } = render(
      <Humbarger HumbargerOpen={() => {}} isHidden={false} />,
    );
    expect(screen.getByTestId("humbarger")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("クリック時にコールバック関数が正しく呼び出されているか", async () => {
    const HumbargerOpen = jest.fn();
    render(<Humbarger HumbargerOpen={HumbargerOpen} isHidden />);
    await userEvent.click(screen.getByTestId("humbarger"));
    expect(HumbargerOpen).toHaveBeenCalledTimes(1);
  });
});
