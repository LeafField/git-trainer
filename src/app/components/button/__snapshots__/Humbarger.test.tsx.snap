// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Humbargerの単体テスト ドロワーメニューが閉じている時に正しくレンダリングされているか 1`] = `
<div>
  <button
    aria-controls="aside"
    aria-expanded="false"
    class="absolute right-1 top-1 z-50 block w-fit lg:hidden"
    data-testid="humbarger"
  >
    <svg
      class="h-11 w-11 text-white hover:text-gray-300"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.879 7.519c1.171-1.025 3.071-1.025 4.242 0 1.172 1.025 1.172 2.687 0 3.712-.203.179-.43.326-.67.442-.745.361-1.45.999-1.45 1.827v.75M21 12a9 9 0 11-18 0 9 9 0 0118 0zm-9 5.25h.008v.008H12v-.008z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;

exports[`Humbargerの単体テスト ドロワーメニューが開いている時に正しくレンダリングされているか 1`] = `
<div>
  <button
    aria-controls="aside"
    aria-expanded="true"
    class="absolute right-1 top-1 z-50 block w-fit lg:hidden"
    data-testid="humbarger"
  >
    <svg
      class="h-11 w-11 text-white hover:text-gray-300"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.75 9.75l4.5 4.5m0-4.5l-4.5 4.5M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </button>
</div>
`;
